src/fluentd/src/fluent-plugin-pgjson/lib/fluent/plugin/out_pgjson.rb

```diff
@@ -167,6 +167,40 @@ def insert_framework(hex_id, time, record)
       end
     end
 
+    def insert_task(hex_id, time, record)
+      # This function try to insert the task snapshot into task history table.
+      # In some cases, the framework controller may have duplicate logs about one task attempt,
+      # or there has been already successful inserted record before.
+      # To handle it, `insert_task` executes a "SELECT" first.
+      # If the uid exists in the table, it will ignore it safely.
+      # Any error should be raised.
+      thread = Thread.current
+      frameworkName = record["objectSnapshot"]["metadata"]["name"][0..31]
+      attemptIndex = record["objectSnapshot"]["metadata"]["annotations"]["FC_FRAMEWORK_ATTEMPT_ID"]
```

Comment:
This is task name instead of frameworkName.

record["objectSnapshot"]["metadata"]["annotations"]["FC_FRAMEWORK_NAME"] is the frameworkName

Reply:
But why your picture is right?
![image](https://user-images.githubusercontent.com/32826762/94359540-89d15500-00da-11eb-9935-c1d3008c6f02.png)

Reply:
because pai task name is "<framework name>-<taskrole name>-<task index>"